src/Symfony/Component/Validator/Constraints/Yaml.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Yaml extends Constraint
+{
+    public const INVALID_YAML_ERROR = '63313a31-837c-42bb-99eb-542c76aacc48';
+
+    protected const ERROR_NAMES = [
+        self::INVALID_YAML_ERROR => 'INVALID_YAML_ERROR',
+    ];
+
+    public $message = 'This value should be valid YAML.';
+    public $flags = 0;
+
+    public function __construct(
+        ?array $options = null,
```

Comment:
We don't need the `$options` parameter anymore as we do no longer support annotations. If we then also make use of constructor property promotion the method can look like this:

```php
public function __construct(
    public string $message = 'This value is not valid YAML.',
    public int $flags = 0,
    ?array $groups = null,
    mixed $payload = null,
) {
    parent::__construct(null, $groups, $payload);
}
```